### PR TITLE
fix(core): Unload components when individually disposed.

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

File: modules/angular2/src/core/application_ref.ts [hunk 1]
```diff
@@ -429,6 +438,15 @@ export class ApplicationRef_ extends ApplicationRef {
     return completer.promise;
   }
 
+  /** @internal */
+  _unloadComponent(ref: ComponentRef): void {
```
Comment (tbosch) [inline modules/angular2/src/core/application_ref.ts]: Have the same for `_loadComponent`.

File: modules/angular2/src/core/application_ref.ts [hunk 2]
```diff
@@ -69,10 +69,15 @@ function _componentProviders(appComponentType: Type): Array<Type | Provider | an
     provide(APP_COMPONENT, {useValue: appComponentType}),
     provide(APP_COMPONENT_REF_PROMISE,
             {
-              useFactory: (dynamicComponentLoader, injector: Injector) => {
+              useFactory: (dynamicComponentLoader, appRef: ApplicationRef_, injector: Injector) => {
```
Comment (vsavkin) [inline modules/angular2/src/core/application_ref.ts]: Can you add a type to dynamicComponentLoader?

Comment (alxhub) [inline modules/angular2/src/core/application_ref.ts]: Done.

**Comment by IgorMinar** [comment]
is this still a thing?

can you rebase please or close if already fixed by another pr

**Comment by IgorMinar** [comment]
@alxhub can you please rebase?

**Comment by naomiblack** [comment]
@alxhub is this PR still relevant?

**Comment by alxhub** [comment]
Fixed by another PR.

**Pull request closed without merging**